File: frontend/src/types/archives.ts
import type { MainProjectRead } from '@/types/projects';

export interface ArchiveCandidateRead {
  closed_at: string;
  main_project_id: string;
  name: string;
  project_no: string;
  sub_project_count: number;
}

export interface ArchiveRunResultRead {
  archived_main_project_count: number;
  archived_sub_project_count: number;
  batch_id: string;
  batch_no: string;
  duration_ms: number;
  main_projects_after: number;
  main_projects_before: number;
}

export interface ArchiveBatchRead {
  archived_main_project_count: number;
  archived_sub_project_count: number;
  batch_no: string;
  created_at: string;
  created_by_id: string | null;
  duration_ms: number;
  finished_at: string;
  id: string;
  started_at: string;
  status: string;
  updated_at: string;
}

export interface ArchiveMainProjectRead {
  archived_at: string;
  batch_id: string;
  closed_at: string | null;
  created_at: string;
  id: string;
  name: string;
  original_id: string;
  project_no: string;
  snapshot: Record<string, unknown>;
  status: string;
  updated_at: string;
}

export interface ArchiveSubProjectRead {
  archived_at: string;
  batch_id: string;
  closed_at: string | null;
  created_at: string;
  id: string;
  name: string;
  original_id: string;
  original_main_project_id: string;
  project_no: string;
  snapshot: Record<string, unknown>;
  status: string;
  updated_at: string;
}

export interface ArchiveBatchListQuery {
  page: number;
  pageSize: number;
}

export interface ArchiveBatchListRead {
  items: ArchiveBatchRead[];
  page: number;
  page_size: number;
  total: number;
}

export interface ArchiveBatchDetailRead {
  batch: ArchiveBatchRead;
  main_projects: ArchiveMainProjectRead[];
  sub_projects: ArchiveSubProjectRead[];
}

export type ArchiveRestoreResultRead = MainProjectRead;

File: backend/app/schemas/archives.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class ArchiveCandidateRead(BaseModel):
    main_project_id: UUID
    project_no: str
    name: str
    closed_at: datetime
    sub_project_count: int


class ArchiveRunResultRead(BaseModel):
    batch_id: UUID
    batch_no: str
    archived_main_project_count: int
    archived_sub_project_count: int
    main_projects_before: int
    main_projects_after: int
    duration_ms: int


class ArchiveBatchRead(BaseModel):
    id: UUID
    batch_no: str
    created_by_id: UUID | None
    status: str
    archived_main_project_count: int
    archived_sub_project_count: int
    duration_ms: int
    started_at: datetime
    finished_at: datetime
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ArchiveMainProjectRead(BaseModel):
    id: UUID
    batch_id: UUID
    original_id: UUID
    project_no: str
    name: str
    status: str
    closed_at: datetime | None
    snapshot: dict[str, object]
    archived_at: datetime
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ArchiveSubProjectRead(BaseModel):
    id: UUID
    batch_id: UUID
    original_id: UUID
    original_main_project_id: UUID
    project_no: str
    name: str
    status: str
    closed_at: datetime | None
    snapshot: dict[str, object]
    archived_at: datetime
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ArchiveBatchListRead(BaseModel):
    items: list[ArchiveBatchRead]
    page: int
    page_size: int
    total: int


class ArchiveBatchDetailRead(BaseModel):
    batch: ArchiveBatchRead
    main_projects: list[ArchiveMainProjectRead]
    sub_projects: list[ArchiveSubProjectRead]
